suites/test_product.py

```diff
@@ -0,0 +1,75 @@
+                                  "locale": "en-US",
+                                  "urlFragment": product_name_uri}
+
+        # Hit the api for create product
+        response = requests.post(self.path_to_product, data=create_product_payload, auth=(username, auth))
+        check_that("The Product is created successfully",
+                   response.status_code, any_of(equal_to(201), equal_to(200)))
+
+        # Hit the api to get id of created product
+        get_product_id_endpoint = self.path_to_product + ".json"
+        product_id_request = requests.get(get_product_id_endpoint)
+        check_that("%s API status code" % get_product_id_endpoint, product_id_request.status_code, equal_to(200))
+        product_id = product_id_request.json()
+        check_that("Product uuid", product_id["jcr:uuid"], is_not_none())
+        check_that("sling:resourceType", product_id["sling:resourceType"], equal_to("pantheon/product"))
+        check_that("jcr:primaryType", product_id["jcr:primaryType"], equal_to("pant:product"))
+        check_that("product name", product_id["name"], equal_to(product_name))
+
+    @lcc.test("Verify product version info : uuid, name, sling:resourceType, jcr:primaryType")
+    def verify_product_version_info(self):
+        path_to_version = self.path_to_product + "/versions/{}".format(constants.product_version)
+        lcc.log_info("Product version being created for the above product: %s" % path_to_version)
+
+        create_version_payload = {"name": constants.product_version,
+                                  "sling:resourceType": "pantheon/productVersion",
+                                  "jcr:primaryType": "pant:productVersion",
+                                  "urlFragment": constants.product_version_uri}
+
+        # Hit the api for create version for the above product
+        response = requests.post(path_to_version, data=create_version_payload, auth=(username, auth))
+        check_that("The Product version is created successfully",
+                   response.status_code, any_of(equal_to(201), equal_to(200)))
+
+        # Hit the api to get id of product version
+        path_to_product_id = self.path_to_product + "/versions/" + constants.product_version + ".json"
+        product_version_id_req = requests.get(path_to_product_id)
+        check_that("%s API status code" % path_to_product_id, product_version_id_req.status_code, equal_to(200))
+        product_version_id = product_version_id_req.json()
+        check_that("Product version uuid", product_version_id["jcr:uuid"], is_not_none())
+        check_that("sling:resourceType", product_version_id["sling:resourceType"], equal_to("pantheon/productVersion"))
+        check_that("jcr:primaryType", product_version_id["jcr:primaryType"], equal_to("pant:productVersion"))
+        check_that("product version", product_version_id["name"], equal_to(constants.product_version))
+
+    @lcc.test("Delete above created product")
```

Comment:
@pbadak this should not be a test, but a tear down step for the test case, the product node should be deleted irrespective that the above tests are successful or not. In addition, add a try catch that the delete should not fail or give an error if the product does not exist given that the above tests failed for some reason.

Reply:
Thanks for your review @anarang . Made changes as per comment .